modules/ROOT/pages/partner-connector-upgrade.adoc

```diff
@@ -0,0 +1,417 @@
+----
+
+[[test-for-java-17-compatibility-directly]]
+===== Test for Java 17 Compatibility Directly
+
+Run your MTF tests to test compatibility of your custom connector against Java 17. 
+
+As mentioned previously, you can have a single build pipeline that runs against all supported Java versions. You can also choose to set up another temporary build pipeline for Java 17 so your main build pipeline does not become unstable. When you are done upgrading to Java 17, you can discard the temporary build pipeline and converge on your main build pipeline.
+
+Use the following bash script to test your custom connector against Java 17: 
+
+[source,bash]
+----
+#!/bin/bash
+RUNTIME_VERSION=4.6.0
+MUNIT_JVM=/Library/Java/JavaVirtualMachines/temurin-17.jdk/Contents/Home/bin/java
+mvn clean
+mkdir target
+mvn verify \
+   -DruntimeProduct=MULE_EE \
+   -DruntimeVersion=$RUNTIME_VERSION \
+   -Dmunit.jvm=$MUNIT_JVM \
+   -Dmule.module.tweaking.validation.skip=true \
+   -Dmule.jvm.version.extension.enforcement=LOOSE > ./target/test.log
+----
+
+Set the path to your JVM installation in the MUNIT_JVM variable, but you must install it yourself. You must also set JAVA_HOME to Java 8. 
+
+You must also ensure the following MTF dependencies are set in your custom connector’s pom.xml file:
+
+* munit 3.1.0
+* munit-extensions-maven-plugin 1.2.0
+* mtf-tools 1.2.0
+* mule-maven-plugin 4.1.0
+* mule-extensions-maven-plugin 1.6.0-rc1
+
+These MTF dependencies have a minimum Mule version of 4.3.0. To ensure your MTF tests do not validate against Mule runtime versions earlier than 4.3.0, you must add this to the munit-plugin configuration of your custom connector’s pom.xml file: 
+
+[source,xml,linenums]
+----
+<configuration>
+	[...]
+<runtimeConfiguration>
+    <discoverRuntimes>
+        <minMuleVersion>${minVersion}</minMuleVersion>
+        <includeSnapshots>false</includeSnapshots>
+        <product>EE</product>
+    </discoverRuntimes>
+</runtimeConfiguration>
+</configuration>
+----
+
+You can only run MTF tests against Java 17 with Mule runtime 4.6.0 and later. For Mule runtime versions earlier than 4.6.0, you can only run MTF tests against Java 8 and Java 11. 
+
+MUnit 3.1 is only compatible with Mule runtime 4.3.0 and later. If your connector is compatible with Mule runtime 4.2.0 and earlier, you must create a legacy profile that overrides the MUnit version.
+
+=== Read Your Tests 
+
+After you run your MTF tests, your build has either of the following outcomes:
+
+* Test failures
++
+You will likely have to make changes to your custom connector code to ensure Java 17 compatibility.
+
+* All tests pass
++
+Either your custom connector does not require any major changes or your test suite is not comprehensive enough. You might want to review your test suite and double-check that your code coverage is good and that your test scenarios and assertions are not excessively naive.
```

Comment:
Not sure what "excessively naive" means. Is there better wording, like "too simple" or something?